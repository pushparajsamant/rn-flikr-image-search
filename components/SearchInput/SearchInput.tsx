import React, {useEffect, useRef, useState} from 'react';
import {StyleSheet, TextInput, TouchableOpacity, View} from 'react-native';
import {
  fontScale,
  horizontalScale,
  verticalScale,
} from '../../assets/styles/scale';
import {faCircleXmark, faSearch} from '@fortawesome/free-solid-svg-icons';
import {FontAwesomeIcon} from '@fortawesome/react-native-fontawesome';

interface Props {
  onSearch?: Function;
  placeholder?: string;
  onFocus?: any | undefined;
  value?: any | undefined;
  clearValue?: any | undefined;
  showClear?: boolean;
  showSubmit?: boolean;
}

const SearchInput: React.FC<Props> = ({
  onSearch,
  placeholder,
  onFocus,
  value,
  clearValue,
  showClear,
  showSubmit,
}) => {
  useEffect(() => {
    // This will ensure that the searched term is visible on both pages. Value is collected from the reducer by parent
    setSearch(value);
    //Focus on the input field only on the Search Screen
    if (onFocus === undefined) {
      handleFocus();
    }
    return () => {};
  }, [value, onFocus]);
  const textInputRef = useRef(null);
  const [search, setSearch] = useState('');
  const handleFocus = () => {
    textInputRef.current && textInputRef.current.focus();
  };
  function handleSearch(searchValue: React.SetStateAction<string>) {
    setSearch(searchValue);
  }
  function handleClearValue() {
    //Clears local value in state and also calls dispatch to clear in reducer
    setSearch('');
    clearValue();
  }

  return (
    <View style={styles.searchBar}>
      <View style={styles.searchViewContainer}>
        <FontAwesomeIcon icon={faSearch} color={'#25C0FF'} size={22} />
        <TextInput
          ref={textInputRef}
          style={styles.searchInput}
          value={search}
          onFocus={onFocus}
          autoCorrect={false}
          placeholder={placeholder}
          onChangeText={searchValue => handleSearch(searchValue)}
        />
        {showClear && search.length > 0 && (
          <TouchableOpacity onPress={() => handleClearValue()}>
            <FontAwesomeIcon icon={faCircleXmark} color={'grey'} />
          </TouchableOpacity>
        )}
      </View>
      {showSubmit && (
        <TouchableOpacity
          style={styles.searchButton}
          onPress={() => {
            if (search.length > 3) {
              onSearch && onSearch(search);
            }
          }}>
          <FontAwesomeIcon icon={faSearch} color={'#03045E'} size={22} />
        </TouchableOpacity>
      )}
    </View>
  );
};
SearchInput.defaultProps = {
  placeholder: 'Search',
  showClear: false,
  showSubmit: false,
  onSearch: () => {},
};
export default SearchInput;
const styles = StyleSheet.create({
  searchViewContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    borderWidth: 1,
    borderColor: '#CAF0F8',
    paddingHorizontal: horizontalScale(10),
    backgroundColor: '#CAF0F8',
    height: verticalScale(50),
    flexGrow: 3,
    borderRadius: horizontalScale(15),
  },
  searchInput: {
    flex: 1,
    marginLeft: horizontalScale(6),
    height: '100%',
    fontFamily: 'Inter',
    fontSize: fontScale(16),
    lineHeight: fontScale(18),
    paddingHorizontal: horizontalScale(16),
  },
  searchBar: {
    flexDirection: 'row',
    paddingHorizontal: horizontalScale(10),
    gap: 5,
    alignItems: 'center',
    marginVertical: verticalScale(10),
  },
  searchButton: {
    height: verticalScale(50),
    width: horizontalScale(50),
    justifyContent: 'center',
    alignItems: 'center',
  },
});
