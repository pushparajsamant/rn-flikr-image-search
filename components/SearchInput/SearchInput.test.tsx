import React from 'react';
import renderer from 'react-test-renderer';
import SearchInput from './SearchInput';
jest.mock('react-native-device-info', () => {
  return {
    hasNotch: () => true,
  };
});
jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));
test('Search Input Component renders correctly with close and submit button', () => {
  const tree = renderer
    .create(
      <SearchInput
        onSearch={() => {}}
        showClear={true}
        showSubmit={true}
        value={''}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test('Search Input Component renders correctly without close and submit button', () => {
  const tree = renderer
    .create(
      <SearchInput
        onSearch={() => {}}
        showClear={false}
        showSubmit={false}
        value={''}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test('Search Input Component renders correctly with placeholder', () => {
  const tree = renderer
    .create(
      <SearchInput
        onSearch={() => {}}
        placeholder={'Search Text'}
        value={''}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
