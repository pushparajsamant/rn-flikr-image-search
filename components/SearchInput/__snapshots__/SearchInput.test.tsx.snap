// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Input Component renders correctly with close and submit button 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "gap": 5,
      "marginVertical": 21.516129032258068,
      "paddingHorizontal": 21.428571428571427,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#CAF0F8",
        "borderColor": "#CAF0F8",
        "borderRadius": 32.14285714285714,
        "borderWidth": 1,
        "flexDirection": "row",
        "flexGrow": 3,
        "height": 107.58064516129033,
        "paddingHorizontal": 21.428571428571427,
      }
    }
  >
    <
      color="#25C0FF"
      icon={
        {
          "icon": [
            512,
            512,
            [
              128269,
              "search",
            ],
            "f002",
            "M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z",
          ],
          "iconName": "magnifying-glass",
          "prefix": "fas",
        }
      }
      size={22}
    />
    <TextInput
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Search"
      style={
        {
          "flex": 1,
          "fontFamily": "Inter",
          "fontSize": 27.906976744186046,
          "height": "100%",
          "lineHeight": 31.3953488372093,
          "marginLeft": 12.857142857142858,
          "paddingHorizontal": 34.285714285714285,
        }
      }
      value=""
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "height": 107.58064516129033,
        "justifyContent": "center",
        "opacity": 1,
        "width": 107.14285714285714,
      }
    }
  >
    <
      color="#03045E"
      icon={
        {
          "icon": [
            512,
            512,
            [
              128269,
              "search",
            ],
            "f002",
            "M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z",
          ],
          "iconName": "magnifying-glass",
          "prefix": "fas",
        }
      }
      size={22}
    />
  </View>
</View>
`;

exports[`Search Input Component renders correctly with placeholder 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "gap": 5,
      "marginVertical": 21.516129032258068,
      "paddingHorizontal": 21.428571428571427,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#CAF0F8",
        "borderColor": "#CAF0F8",
        "borderRadius": 32.14285714285714,
        "borderWidth": 1,
        "flexDirection": "row",
        "flexGrow": 3,
        "height": 107.58064516129033,
        "paddingHorizontal": 21.428571428571427,
      }
    }
  >
    <
      color="#25C0FF"
      icon={
        {
          "icon": [
            512,
            512,
            [
              128269,
              "search",
            ],
            "f002",
            "M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z",
          ],
          "iconName": "magnifying-glass",
          "prefix": "fas",
        }
      }
      size={22}
    />
    <TextInput
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Search Text"
      style={
        {
          "flex": 1,
          "fontFamily": "Inter",
          "fontSize": 27.906976744186046,
          "height": "100%",
          "lineHeight": 31.3953488372093,
          "marginLeft": 12.857142857142858,
          "paddingHorizontal": 34.285714285714285,
        }
      }
      value=""
    />
  </View>
</View>
`;

exports[`Search Input Component renders correctly without close and submit button 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "gap": 5,
      "marginVertical": 21.516129032258068,
      "paddingHorizontal": 21.428571428571427,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#CAF0F8",
        "borderColor": "#CAF0F8",
        "borderRadius": 32.14285714285714,
        "borderWidth": 1,
        "flexDirection": "row",
        "flexGrow": 3,
        "height": 107.58064516129033,
        "paddingHorizontal": 21.428571428571427,
      }
    }
  >
    <
      color="#25C0FF"
      icon={
        {
          "icon": [
            512,
            512,
            [
              128269,
              "search",
            ],
            "f002",
            "M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z",
          ],
          "iconName": "magnifying-glass",
          "prefix": "fas",
        }
      }
      size={22}
    />
    <TextInput
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Search"
      style={
        {
          "flex": 1,
          "fontFamily": "Inter",
          "fontSize": 27.906976744186046,
          "height": "100%",
          "lineHeight": 31.3953488372093,
          "marginLeft": 12.857142857142858,
          "paddingHorizontal": 34.285714285714285,
        }
      }
      value=""
    />
  </View>
</View>
`;
